Added an example plot output

iamprayush opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests .-->
#15327

#### Brief description of what is fixed or changed
Added a plot output to the append example in plot.py.

#### Other comments


#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* plotting
  * added an example output to the sphinx documentation
<!-- END RELEASE NOTES -->

iamprayush committed:
Added output for append example

Added plot output via matplotlib plot directive.

iamprayush committed:
Added the append plot output

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -237,6 +237,16 @@ def append(self, arg):
+	
+	    .. plot::
+
+	       >>> from sympy import symbols
```
incorrect indentation

iamprayush replied:
this is the formatting with which I got a proper output. With any other formatting, I was getting an error. If you could tell me the correct indentation, that'd be helpful.

moorepants replied:
all indentation of the `>>>` lines should be the same

iamprayush replied:
Alright, I've corrected the indentation. What shall I do about the redundant code?

moorepants commented:
Please post a screenshot of the rendered version.

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -237,6 +237,16 @@ def append(self, arg):
+           >>> p1 = plot(x*x, show = False)
+           >>> p2 = plot(x, show = False)
+           >>> p1.append(p2[0])
+           >>> p1.show()	
```
You have the same code twice. Please only have 1 example and 1 plot.

iamprayush replied:
I had to add the show = False in the code to prevent multiple plots being shown, so if it's ok for "show = False" showing up in the example code, I can make necassary changes to remove the redundant code.

And here is the screenshot:
![ss1](https://user-images.githubusercontent.com/35144226/46560298-c796ab00-c910-11e8-8b48-f1eee15f2be3.png)

iamprayush committed:
Corrected the indentation

iamprayush committed:
Indentation correction

The indentation seems fine while editing but changes in the output and the preview.

moorepants commented:
This is looking good. Make sure the doctests pass and we will see what Travis CI tells us.

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -237,6 +237,16 @@ def append(self, arg):
+           >>> x = symbols('x')
+           >>> p1 = plot(x*x, show = False)
+           >>> p2 = plot(x, show = False)
+           >>> p1.append(p2[0])
```
These are still misaligned. All `>>>` should be left aligned with the "p" in plot.

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -237,6 +237,16 @@ def append(self, arg):
         Plot object containing:
         [0]: cartesian line: x**2 for x over (-10.0, 10.0)
         [1]: cartesian line: x for x over (-10.0, 10.0)
```
Please edit the above example, don't add a new one.

iamprayush committed:
Removed redundant code.

iamprayush commented:
I don't understand. Why does Travis CI show that doctest failed when the run perfectly fine on my local machine?

iamprayush commented:
This is the error that the tests are showing:
![screenshot from 2018-10-06 09-51-55](https://user-images.githubusercontent.com/35144226/46567243-8a9cd980-c94d-11e8-92a7-47365fcb4284.png)
Does it mean that it expects the two plots to be outputted and since I've disabled their output (show = False), it is giving me an error? If this is the issue then maybe it could be fixed by writing the same code again.

iamprayush committed:
Trying again with minor changes in formatting

iamprayush committed:
Trying again with the redundant code.

This is just to test if my changes to the code block were causing the errors in tests.

iamprayush committed:
Trying to fix error produced by Travic CI

All tests are passed on my local machine. I explicitly tested out the code_quality test and all other tests. They all passed.

iamprayush commented:
@moorepants 
Ok, so finally all checks have passed! And I've checked the output which looks proper (shown in the screenshot uploaded above). Please merge this now, so I can start adding outputs to other examples.

iamprayush commented:
And, btw, I've also removed the redundant code and these checks have passed on the code without any repetitions so everything should be perfect now.

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -227,16 +227,21 @@ def append(self, arg):
-        [0]: cartesian line: x**2 for x over (-10.0, 10.0)
-        [1]: cartesian line: x for x over (-10.0, 10.0)
+        .. plot::
+           :context:
```
Why is `:context:` needed? It seems everything is self contained here.

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -227,16 +227,21 @@ def append(self, arg):
+           >>> from sympy.plotting import plot
+           >>> x = symbols('x')
+           >>> p1 = plot(x*x, show = False)  # doctest: +SKIP
+           >>> p2 = plot(x, show = False)  # doctest: +SKIP
```
No spaces around the `=` between show and false. See PEP8 for guidelines on style.

Why do these two lines need to be skipped?

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -227,16 +227,21 @@ def append(self, arg):
+           >>> x = symbols('x')
+           >>> p1 = plot(x*x, show = False)  # doctest: +SKIP
+           >>> p2 = plot(x, show = False)  # doctest: +SKIP
+           >>> p1.append(p2[0])  # doctest: +SKIP
```
Why does this need to be skipped?

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -227,16 +227,21 @@ def append(self, arg):
+           >>> p1.show()  # doctest: +SKIP
+           Plot object containing:  # doctest: +SKIP
+           [0]: cartesian line: x**2 for x over (-10.0, 10.0)  # doctest: +SKIP
+           [1]: cartesian line: x for x over (-10.0, 10.0)  # doctest: +SKIP
```
Don't skip this test. I was working before. Leave in the call to print p1 before plotting.

moorepants commented on sympy/plotting/plot.py:
```diff
@@ -227,16 +227,21 @@ def append(self, arg):
+           >>> p1 = plot(x*x, show = False)  # doctest: +SKIP
+           >>> p2 = plot(x, show = False)  # doctest: +SKIP
+           >>> p1.append(p2[0])  # doctest: +SKIP
+           >>> p1.show()  # doctest: +SKIP
```
This line can be skipped if it doesn't make the below output.

moorepants commented:
This is what I get:
![selection_273](https://user-images.githubusercontent.com/276007/46571951-e443da80-c932-11e8-9619-88e9597c237b.png)

It isn't clear to me why the plots show separately.

iamprayush commented:
Ok, so I made the changes and the code looks something like this:
![screenshot from 2018-10-06 19-14-29](https://user-images.githubusercontent.com/35144226/46572011-b6937b80-c99c-11e8-844d-6bca47ff3273.png)
All the tests passed for this file and the output looks like this for me:
![screenshot from 2018-10-06 19-17-37](https://user-images.githubusercontent.com/35144226/46572018-cd39d280-c99c-11e8-834d-7b1cad8e5739.png)

The reason I was trying to skip doctests for some lines was that travis CI was giving an error (as shown by the screenshot I uploaded previously), even though no errors were observed when I ran the tests locally.

moorepants commented:
>  even though no errors were observed when I ran the tests locally.

I've noticed this too when I was working on the beam module plots. Not sure why that is.

If `p1.show()` creates the output: "Plot object containing.." then you can get rid of `>>> p1`.

moorepants commented:
Maybe the doc tests are skipped when run locally and not on Travis.

iamprayush commented:
> > even though no errors were observed when I ran the tests locally.

> If `p1.show()` creates the output: "Plot object containing.." then you can get rid of `>>> p1`.
It probably doesn't, right? Or else there wouldn't have been errors in travis upon removing '>>>p1'

iamprayush commented:
So, should I commit these changes? Without skipping the doctest on some lines? I believe there will be errors then.

moorepants commented:
Seems like it does doctest it:

```
(sympy-dev) moorepants@garuda:sympy(master)$ bin/doctest -vv sympy/plotting/plot.py
========================================== test process starts ==========================================
executable:         /home/moorepants/miniconda3/envs/sympy-dev/bin/python  (3.5.5-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.11.2
hash randomization: on (PYTHONHASHSEED=353590689)

sympy/plotting/plot.py[8] 
sympy.plotting.plot.check_arguments ok
sympy.plotting.plot.plot3d_parametric_surface ok
sympy.plotting.plot.plot3d ok
sympy.plotting.plot.plot3d_parametric_line ok
sympy.plotting.plot.plot_parametric ok
sympy.plotting.plot.plot ok
sympy.plotting.plot.Plot.extend ok
sympy.plotting.plot.Plot.append ok                                                                   [OK]
```

moorepants commented:
Yes you should commit so that we can debug on Travis if there is an error.

iamprayush commented:
Alright, I'll commit these changes. Hopefully, all checks will pass.

moorepants commented:
I also don't understand why I get different output locally than you do.

iamprayush committed:
No doctests skipped now.

iamprayush commented:
> I also don't understand why I get different output locally than you do.

I was also getting similar output with two different plots but when I added the 'show=False', I only got the appended single plot. Not sure why yours is different.

iamprayush commented:
Ok, see? This is the error we're getting now:
![screenshot from 2018-10-06 19-59-44](https://user-images.githubusercontent.com/35144226/46572408-daf25680-c9a2-11e8-9edf-2811c32af9d4.png)
Maybe I should swap the positions of p1 and p1.show() or maybe just add the skip doctest.

iamprayush committed:
swapped positions to try to avoid errors

moorepants commented:
You should either deleted the `>>> p1` line or add its result under it. Like my code I posted.

moorepants commented:
```pycon
           >>> from sympy import symbols
           >>> from sympy.plotting import plot
           >>> x = symbols('x')
           >>> p1 = plot(x*x, show=False)
           >>> p2 = plot(x, show=False)
           >>> p1.append(p2[0])
           >>> p1
           Plot object containing:
           [0]: cartesian line: x**2 for x over (-10.0, 10.0)
           [1]: cartesian line: x for x over (-10.0, 10.0)
           >>> p1.show()
```

iamprayush commented:
Alright, @moorepants, this should be good enough to merge now. All the checks have passed.

moorepants commented:
Great! Looks perfect. Thanks for the first contribution. Now that you know how to set them up, feel free to add more plots.

iamprayush commented:
Thanks for being such a patient guide. This was really helpful and I shall add more plots soon.
